samples/snippets/src/main/java/pubsub/PublishWithFlowControlExample.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pubsub;
+
+// [START pubsub_publisher_flow_control]
+
+import com.google.api.core.ApiFuture;
+import com.google.api.core.ApiFutures;
+import com.google.api.gax.batching.BatchingSettings;
+import com.google.api.gax.batching.FlowControlSettings;
+import com.google.api.gax.batching.FlowController.LimitExceededBehavior;
+import com.google.cloud.pubsub.v1.Publisher;
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.TopicName;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+public class PublishWithFlowControlExample {
+  public static void main(String... args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-project-id";
+    String topicId = "your-topic-id";
+
+    publishWithFlowControlExample(projectId, topicId);
+  }
+
+  public static void publishWithFlowControlExample(String projectId, String topicId)
+      throws IOException, ExecutionException, InterruptedException {
+    TopicName topicName = TopicName.of(projectId, topicId);
+    Publisher publisher = null;
+    List<ApiFuture<String>> messageIdFutures = new ArrayList<>();
+
+    try {
+      // Configure how many messages the publisher client can hold in memory
+      // before publishing succeeds, and what to do when messages exceed the limit.
+      FlowControlSettings flowControlSettings =
+          FlowControlSettings.newBuilder()
+              // Block more messages from being published when the limit is reached. The other
+              // options are Ignore (or continue publishing) and ThrowException (or error out).
+              .setLimitExceededBehavior(LimitExceededBehavior.Block)
+              .setMaxOutstandingRequestBytes(10 * 1024 * 1024L) // 10 MiB
+              .setMaxOutstandingElementCount(100L) // 100 messages
+              .build();
+
+      // By default, messages are not batched.
+      BatchingSettings batchingSettings =
+          BatchingSettings.newBuilder().setFlowControlSettings(flowControlSettings).build();
+
+      publisher = Publisher.newBuilder(topicName).setBatchingSettings(batchingSettings).build();
+
+      // Publish 1000 messages in quick succession to trigger publisher flow control.
```

Comment:
I wouldn't say "to trigger flow control." It's possible that flow control will be triggered, but it's also entirely possible it won't be if messages get out quickly enough. Either way, the fact that flow control was triggered isn't something that will be easily visible to the user. Maybe say "Rapidly publishing 1000 messages in a loop may be constrained by flow control."

Reply:
Updated. True, the user will only know for sure if they turn on logging.